docs/en/custom-elements/introduction.md

```diff
@@ -0,0 +1,164 @@
+
+Dojo widgets can be built as custom elements using @dojo/cli-build-widget. For this example, we're going to use a simple email subscription widget.
+
+> src/widgets/EmailSignup.tsx
+
+```tsx
+import { tsx, create } from '@dojo/framework/core/vdom';
+import * as css from './styles/EmailSignup.m.css';
+import icache from '@dojo/framework/core/middleware/icache';
+
+export interface EmailSignupProperties {
+	title?: string;
+	onSignUp?: () => void;
+	signUpBonusDate?: Date;
+}
+
+const factory = create({ icache }).properties<EmailSignupProperties>();
+
+export default factory(function EmailSignup({ middleware: { icache }, properties }) {
+	const { title, signUpBonusDate } = properties();
+
+	const signedUp = icache.getOrSet('signedUp', false);
+	const signUp = () => {
+		const { onSignUp } = properties();
+
+		// post email address to a server
+		icache.set('signedUp', true);
+		onSignUp && onSignUp();
+	};
+
+	return (
+		<div key="root" classes={css.root}>
+			<div key="title" classes={css.title}>
+				{title || 'Subscribe Now'}
+			</div>
+			{signedUp ? (
+				<div key="confirmation" classes={css.thanks}>
+					Thank you for signing up!
+				</div>
+			) : (
+				<div key="signup" classes={css.signup}>
+					{signUpBonusDate && <div>Sign up by {signUpBonusDate.toLocaleDateString()}</div>}
+					<input type="text" />
+					<button onclick={signUp}>Sign Up</button>
+				</div>
+			)}
+		</div>
+	);
+});
+```
+
+To build a widget as a custom element, add a `build-widget` property to your `.dojorc` file and include the path to any widgets you want to expose as custom elements.
+
+> .dojorc
+
+```json
+{
+	"build-widget": {
+		"prefix": "ref",
+		"widgets": ["src/widgets/EmailSignup"]
+	}
+}
+```
+
+With this configuration in place, build the widgets using the Dojo CLI.
+
+```shell
+$ dojo build widget
+```
+
+The `output/dist/` directory now contains the built custom elements.
+
+| Filename                  | Description                                                                                                                                                                             |
+| ------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `ref-email-signup.js`     | This file includes all of the code required to hydrate your custom element. Loading this script in your HTML file is what makes your Dojo widget come alive as a custom element.        |
+| `ref-email-signup.js.map` | A JavaScript source map file for making debugging easier. These files are used only during debugging and do not need to be uploaded to a production environment.                        |
+| `ref-email-signup.css`    | The CSS used by the widget. Note that this includes the styles only for this one widget. If you are using a theme, you will need to separately include the theme CSS in your HTML page. |
```

Comment:
I'm not sure this makes sense as a table. Also for the theme you have to include the theme js too right? we should probably have a complete separate section on theme maybe?